packages/components/src/IconsProvider/IconsProvider.component.js

```diff
@@ -11,9 +40,13 @@ import talendIcons, { filters } from '@talend/icons/dist/react';
  * @example
 <IconsProvider />
  */
-function IconsProvider({ defaultIcons, icons }) {
+function IconsProvider({ defaultIcons, icons, getIconHref }) {
 	const iconset = Object.assign({}, defaultIcons, icons);
 	const ids = Object.keys(iconset);
+	context.ids = ids;
+	if (getIconHref && context.get !== getIconHREF) {
```

Comment:
I don't get this part.
Why do we configure anything in the IconProvider render ? And here you set `context.get = getIconHREF;` but in `getIconHREF` you call `context.get`. Infinite loop ?

Let's discuss about this part